src/server/identity/server/dex_api.go

```diff
@@ -0,0 +1,334 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"sync"
+
+	dex_api "github.com/dexidp/dex/api/v2"
+	dex_server "github.com/dexidp/dex/server"
+	dex_storage "github.com/dexidp/dex/storage"
+	"github.com/pkg/errors"
+	logrus "github.com/sirupsen/logrus"
+
+	"github.com/pachyderm/pachyderm/src/client/identity"
+)
+
+// dexAPI wraps an api.DexServer and extends it with CRUD operations
+// that are currently missing from the dex api. It also handles lazily
+// instantiating the api and storage on the first request.
+type dexAPI struct {
```

Comment:
Is this the file that implements hot-reloading of the dex config? How does that happen?